Lesson4/src/com/home/MilitaryOffice.java

```diff
@@ -0,0 +1,27 @@
+package com.home;
+
+import com.home.model.Address;
+import com.home.model.Person;
+
+public class MilitaryOffice {
+    private PersonRegistry registry;
+
+    public MilitaryOffice(PersonRegistry registry) {
+        this.registry = registry;
+    }
+
+    public void showSoldiers(Address address) {
+        System.out.println("Welcome to the Army!");
+        for (Person person : registry.getCitizens()) {
```

Comment:
this is almost correct. The problem is - you are requesting all the citizens from the registry, but you need to get only those who are living in given address. So instead of getCitizens you need to call getPeople(address), and registry will give you the list of citizens that live by given address